### DISCUSSION TITLE
Pre-processing ERA5-Land foring data

### DISCUSSION THREAD

**Discussion by anlei-code** [NCAR/noahmp#163] [Q&A] [answered]
i have downloaded the previous data,but it dosen't find the varibles.,like var169 .
![图片](https://github.com/user-attachments/assets/993c1cdf-4240-475f-be60-f5f3a71cb1b4)
![图片](https://github.com/user-attachments/assets/ad7741bf-d58c-4158-b9e8-0cd46154b1ee)
![图片](https://github.com/user-attachments/assets/7322fb9b-7162-4540-bed0-46b176b9d639)
![图片](https://github.com/user-attachments/assets/09d20fcd-b54b-4089-8c56-62a453ac6488)

**Comment by cenlinhe** [comment]
have you checked if your PART 1 process was successfully completed and generated those output files? Looks like those variable output files are missing

**Reply by anlei-code** [reply]
I have check the part one outputs,all datas is the date of 2000,but lack of 1999,and from the part1 code ,it only process the INFILE,so i think that it right,the error maybe is the part2:
cdo -s -selvar,var169 -seldate,${predate}T23:00:00,${predate}T23:00:00 $PREVIOUS_FILE OUTPUTS/SSRD/ERA5_ACSSRD_"$preyear""$premonth""$preday"23
cdo -s -selvar,var175 -seldate,${predate}T23:00:00,${predate}T23:00:00 $PREVIOUS_FILE OUTPUTS/STRD/ERA5_ACSTRD_"$preyear""$premonth""$preday"23
cdo -s -selvar,var228 -seldate,${predate}T23:00:00,${predate}T23:00:00 $PREVIOUS_FILE OUTPUTS/TP/ERA5_ACTP_"$preyear""$premonth""$preday"23
this code can't extract the vars,so it may be the cdo version is not compatibale?

**Reply by anlei-code** [reply]
[process.txt](https://github.com/user-attachments/files/18032304/process.txt)
this is my setting

**Reply by cenlinhe** [reply]
it looks like the part 2 error occurred due to not finding the file for 1999123123 for those variables. Is your file name from part 1 correct with the suffix of "_1999123123"?

**Reply by cenlinhe** [reply]
you may want to try again the part 1 process by re-downloading the ERA5-Land data file for the entire day of 1999-12-31 instead of just one timestamp for 23:00:00 of that day.

**Reply by cenlinhe** [reply]
@CharlesZheZhang what do you think may be the issue here?

**Reply by anlei-code** [reply]
I also have re-downloading the ERA5-Land data file for the entire day of 1999-12-31 ,change the cdoe to "cdo -s -selvar,var169 -seldate,${predate}T00:00:00,${predate}T23:00:00 $PREVIOUS_FILE OUTPUTS/SSRD/ERA5_ACSSRD_"$preyear""$premonth""$preday"23",but it not useful.The same problem appeared

**Reply by anlei-code** [reply]
I want to know whether the version of cdo can effect it.
I down the cdo from the terminal "sudo apt-get install cdo"

**Reply by anlei-code** [reply, marked as answer]
I have solve the problem,becasue of the version of cdo 。from terminal download the cdo may be the newest,i down load the cdo version 2.4.0 it can run.

**Comment by CharlesZheZhang** [comment]
Hi @anlei-code , great that you can solve this issue already. Some tips on using cdo command:
cdo -infov -selname,varname <infile>
to check variable information on input data.
